### PR TITLE
Adds the Staff of Rot and Rotten Invocations

### PR DESCRIPTION
**Pull request by tralezab**
:cl:
add: The wizard now gets the staff of rot! It ruins hygiene, spreads miasma, and rots the spells in your brain!
/:cl:
The bolt of rot itself is not damaging, but with how much miasma you can spread and hygiene ruining, it's chaotic.

The rotten invocation is what your spells turn into, it is a fireball that shoots bolts of rot instead of fireballs. I'm planning for there to be multiple kinds of rot spells that are worse versions of popular wizard spells. Jaunt, immovable rod form, who knows.

In a round where the wizard summons magic, the wizard can use the staff of rot to spread magic rot like wildfire. Getting hit by one turns you into a rot user. Wizard is a chaotic game mode and I can't wait to see how fast the rot spreads, and how the rest of the amateur magicians try to hide from the rotten, diseased spell users.

other ideas i'm probably going to leave out of this pr: Lavaland ruin that, with your permission, rots your spells. Great way to neuter a wizard if you want to put in the effort.
Wizard spell that counts as a rot spell as in it cant be removed, but is fairly powerful

### CODE REVIEW

**Comment by tralezab** [comment]
Sprites done by me, sorry if they are bad.
I still need to record some sounds for the spell firing. After that and some more rot spells, this should be ready for merge

**Comment by carshalash** [comment]
Shouldn't it have a different invocation?

**Comment by tralezab** [comment]
What do you mean

**Comment by carshalash** [comment]
Unless i'm misreading it, isn't it just using the fireball Invocation?

**Comment by tralezab** [comment]
Oh yeah whoops I definitely have to change that

**Comment by Shadowflame909** [comment]
"Stinky Time"

Wizard rounds will be changed forever...

**Comment by PKPenguin321** [comment]
Pretty cool, make adjust_disgust(60) to induce a good amount of vomiting

**Comment by tralezab** [comment]
Great idea, it's in my next commit

**Comment by OverlyGeneric** [comment]
What is this 40k? We Nurgle now boys?

**Comment by tralezab** [comment]
image
image

Any more suggestions for nurgle-ified spells?

**Comment by tralezab** [comment]
Yeah, I finished a rotten version of touch spells in the form of the diseased touch that gives a disease (along with the usual effects of the rotten spells)

remember that these spells are going to be spreading pretty quickly as anyone hit by them in a summon magic round becomes a rot user, so they have to be fairly low power. their power comes from the fact that everyone is using them, and they all create bits of miasma and makes everyone unhygienic, contributing to more miasma. More miasma: more diseases, and that's when the culling begins, rot user or not.

**Comment by Shadowflame909** [comment]
Since its a wizard round and they end fast anyways, have Nurglefied Ei-Nath give a very bad disease, like a non-spreading Cuban Pete disease.

**Comment by ATH1909** [comment]
Don't let spellbooks (or staves) of rot spawn from summon magic; it'll basically turn every summon magic round into a rot-fest and ruin the fun of everyone who wanted to actually USE the cool magic spells they got (this is especially disappointing for the wizard, who likely purchased summon magic to cause magical chaos/fun, not make everything stink).

Let wizards buy staves, wands, and/or spellbooks of rot, though- maybe even for free (except for the spellbook one, which should cost 1 point so as not to be a free pick), since they can so easily be turned against the wizard.

**Comment by ATH1909** [comment]
> i'm not actually sure how potent the spreading of nurgle's spells are, I still like it as a force the wizard can go into for chaos, but something the wizard should fear if they are not building for it. so i'll see how i can make the spellbook a rare drop from summon magic.

Also, I think I edited my post and added something to it after you made your post. Just letting you know in case you missed it.

**Comment by tralezab** [comment]
yeah, those are fair points, i'll go with your suggestion

**Comment by OverlyGeneric** [comment]
new idea: make rot bolt a ball of snot. make casting sound like a sneeze.

**Comment by tralezab** [comment]
@Therapist1 I have something really close, it doesn't stun or have any immediate effect but it gives a powerful disease. I think it's actually more powerful than your suggestion because from some short testing there is going to be a lot of vomiting

**Comment by MrDoomBringer** [comment]
forcewall -> wall that makes you barf when you walk through it

**Comment by tralezab** [comment]
@MrDoomBringer omg GENIUS. it'll be in the next commit.

**Comment by tralezab** [comment]
image

**Comment by Kromgar** [comment]
image

This pleases me greatly. How about spellcasting for a rot wizard also eats up your hunger/stamina. So you need to gorge yourself to continue casting at a certain point? 

How about rot version of the hulk/eyelaser spell transforms you into a Great Rotted One? Instead of eyelasers you can projectile vomit at those nearby and i guess keep super strength as flabby corpulent strikes?

rot jaunt could be you can vomit walk or filth walk something like that. 

Rot Repulse could be you release a noxious gas that bows back anyone right next to you? where did this gas come from?  Why must you ask questions is the only answer. 

Magic Missile becomes "Summon Swarm" a swarm of flies flies forth and bites your enemies perhaps laying eggs in their flesh where more flies could propogate from if not treated.

**Comment by tralezab** [comment]
That's a big post with a very nice nurgle at the top, i'll go down the list

* People will already be hungry because getting hit makes you disgusted, the price for using the spells is that they ruin the station and can be turned against you

* Corpulent form sounds great. Didn't someone just add the smelly mutation?

* Just thinking, a jaunt form that dumps miasma wherever you go (thus letting people see where you jaunt, unless the area is fully miasma'd) sounds like a fun spell

* A toxic smoke spell is very in theme with what i'm doing, i'll include it

* I could make a locust swarm (bees renamed) that injects a reagent that makes you more disgusted and unhygienic, overdose gives you the beesease

**Comment by ATH1909** [comment]
> forcewall -> wall that makes you barf when you walk through it

Or maybe something that makes a temporary gluttony ruin wall- you need to be fat to walk through it (which can be hard if you're constantly vomiting).

**Comment by ATH1909** [comment]
Rotten smoke spells (including the rotten version of chaplain's cloudchaser spell) should make clouds of miasma instead of clouds of smoke.

Maybe you could add a spell that has something to do with/is like the viral bonding surgery?

What should a rotten version of bind soul do? Make your respawn form a zombie instead of a skeleton?

**Comment by tralezab** [comment]
<details>
  <summary>WIP sprite</summary>

  image

</details>

**Comment by pizzatiger** [comment]
Make it so the bind soul spell turns you into a spell casting morph once you die

**Comment by Shadowflame909** [comment]
I hope such a spell would fix the currently broken morphs.


They're slow in both forms, and it obliterates their hit and run tactics which allows them to eat important things and get away.

They can no longer get away.

**Comment by Shdorsh** [comment]
> 
> 
> Don't let spellbooks (or staves) of rot spawn from summon magic; it'll basically turn every summon magic round into a rot-fest and ruin the fun of everyone who wanted to actually USE the cool magic spells they got from their spellbooks (this is/would be especially disappointing for the wizard, who likely purchased summon magic to cause magical chaos/fun, not make everything stink).
> 
> This will also serve as a severe disincentive for wizards to choose summon magic, even for the memes/fun- one blow from a rot spell, and they lose their wizard powers PERMANENTLY (even if they're a lich).
> 
> Let wizards buy staves, wands, and/or spellbooks of rot, though- maybe even for free (except for the spellbook one, which should cost 1 point so as not to be a free pick), since they can so easily be turned against the wizard.
> 
> This PR gets a thumbs down from me solely because of the addition of a spellbook of rot to the list of summon magic items (the _common_ magic item list, no less, not even the special/rare one that other round-defining items (like staves of change) are in). If this becomes an opt-in thing for a wizard instead of an unwanted potential tumor on the side of every summon magic round, I'll change my downvote into an upvote.
> 
> Perhaps a "summon spells" ritual that gives everyone on the station random non-rot wizard spells (modified to be robeless) would be a nice addition for wizards who do want to do a rot plague gimmick (by buying a staff of rot and shooting people with it). Right now, only about 2/5 of the station typically gets access to spellbooks in summon magic rounds, which makes for a sort of "meh" zombie-apocalypse-but-with-wizards.

Doesn't matter when you go all out blood demon, though.

**Comment by Shdorsh** [comment]
You could also make a spell that works like a combo, that increases miasma amount on casting any other spell. You'd want to modify the parent spell type's way it works, adding ..() and then locating the modified spell and triggering it.

**Comment by plapatin** [comment]
brap

**Comment by tralezab** [comment]
@Shdorsh wish you were in this thread earlier its not a bad idea, I just already have a full roster of spells plus the corpulent demon's spells

File: code/game/objects/items/granters.dm [hunk 1]
```diff
@@ -150,6 +150,20 @@
...
 	qdel(src)
 
+/obj/item/book/granter/spell/rot
+	spell = /obj/effect/proc_holder/spell/aimed/canker
+	spellname = "rotten invocation"
+	icon_state ="bookrot"
+	desc = "This book smells..."
+	remarks = list("Gross...", "By god, why are the pages slimy?", "Accounting for crosswinds... really?", "I'm going to have to wash my hands after this.", "I'm gonna be sick...", "So my spellrot turns their spells into spellrot, which can turn my spells...", "Spellrot? Seems pretty niche.")
+
+/obj/item/book/granter/spell/rot/recoil(mob/user)
```
Comment (Cyberboss) [inline code/game/objects/items/granters.dm]:
```suggestion
/obj/item/book/granter/spell/rot/recoil(mob/living/user)
```

File: code/game/objects/structures/traps.dm [hunk 2]
```diff
@@ -128,6 +128,33 @@
...
 
+/obj/structure/trap/rot
+	name = "rot trap"
+	desc = "A trap that will spread death and decay. You'd better avoid it."
+	icon_state = "trap-rot"
+
+/obj/structure/trap/rot/trap_effect(mob/living/M)
+	to_chat(M, "<span class='danger'><B>The ground explodes into a cloud of rotting death!</B></span>")
+	M.adjustToxLoss(15)
+	var/turf/T = get_turf(src.loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/traps.dm]:
```suggestion
	var/turf/T = get_turf(src)
```

File: code/game/objects/structures/traps.dm [hunk 3]
```diff
@@ -128,6 +128,33 @@
...
+	if(T)
+		var/datum/effect_system/smoke_spread/bad/smoke = new
+		smoke.set_up(4, M.loc)
+		smoke.start()
+		playsound(M.loc, 'sound/effects/bamf.ogg', 50, 2)
```
Comment (Cyberboss) [inline code/game/objects/structures/traps.dm]:
```suggestion
		playsound(T, 'sound/effects/bamf.ogg', 50, 2)
```

File: code/game/objects/structures/traps.dm [hunk 4]
```diff
@@ -128,6 +128,33 @@
...
+	desc = "A trap that will spread death and decay. You'd better avoid it."
+	icon_state = "trap-rot"
+
+/obj/structure/trap/rot/trap_effect(mob/living/M)
+	to_chat(M, "<span class='danger'><B>The ground explodes into a cloud of rotting death!</B></span>")
+	M.adjustToxLoss(15)
+	var/turf/T = get_turf(src.loc)
+	if(T)
+		var/datum/effect_system/smoke_spread/bad/smoke = new
+		smoke.set_up(4, M.loc)
```
Comment (Cyberboss) [inline code/game/objects/structures/traps.dm]:
```suggestion
		smoke.set_up(4, T)
```

File: code/modules/mob/mob.dm [hunk 5]
```diff
@@ -689,6 +689,28 @@
 			mob_spell_list -= S
 			qdel(S)
 
+/mob/proc/rot_mind() //most rotten spells also swap the victims spells for more rot spells, this proc handles that.
```
Comment (Cyberboss) [inline code/modules/mob/mob.dm]: This should be a mind proc

File: code/modules/projectiles/projectile/magic.dm [hunk 6]
```diff
@@ -445,6 +445,40 @@
...
+	flag = "magic"
+
+/obj/item/projectile/magic/rot/on_hit(target)
+	. = ..()
+	//bolt of rot does a couple things:
+	//dumps miasma wherever it hits
+	//fucks hygiene pretty hard
+	//makes the target feel disgusted
+	//rot_mind is called, see spell.dm for it's effects
+	var/turf/T = get_turf(src.loc)
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/magic.dm]:
```suggestion
	var/turf/T = get_turf(src)
```

File: code/modules/spells/spell_types/rot_spells.dm [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+/obj/effect/forcefield/rotwall
+	name = "Rotten Flesh"
+	desc = "A barrier of pestilence. You feel as if you could go through, but dark energies emanate from it..."
+	icon_state = "rotwall"
+	var/mob/wizard
+
+/obj/effect/forcefield/rotwall/Initialize(mapload, mob/summoner)
+	. = ..()
+	wizard = summoner
+	var/turf/T = get_turf(src.loc)
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/rot_spells.dm]:
```suggestion
	var/turf/T = get_turf(src)
```

File: code/modules/spells/spell_types/rot_spells.dm [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+	if(T)
+		var/datum/gas_mixture/stank = new
+		ADD_GAS(/datum/gas/miasma, stank.gases)
+		stank.gases[/datum/gas/miasma][MOLES] = 100 //300 moles in total
+		T.assume_air(stank)
+		T.air_update_turf()
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/rot_spells.dm]: This is duplicated twice, put it in it's own proc

File: code/modules/spells/spell_types/rot_spells.dm [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+		user.set_species(/datum/species/fly)
+		to_chat(user, "<span class='notice'>Your form becomes that of a fly. Recast to gain powers!</span>")
+	else
+		..()
+
+/*
+/obj/effect/proc_holder/spell/corpulent_summon
+	name = "Rotten Invocation: Summon Corpulent Demon"
+	desc = "Summons a Corpulent Demon. You must attract him to the realm with bodies around you for him to feast."
+*/
```
Comment (Cyberboss) [inline code/modules/spells/spell_types/rot_spells.dm]: Remove commented code

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/spells/spell_types/rot_spells.dm (hunk 8 above)
Comment (tralezab) [reply code/modules/spells/spell_types/rot_spells.dm]: Which file would be best to do that?

Comment (tralezab) [reply code/modules/spells/spell_types/rot_spells.dm]: Which file would be best to do that?

Comment (tralezab) [reply code/modules/spells/spell_types/rot_spells.dm]: actually, i'm fairly sure atmos_spawn_air is a proc that does exactly what I want for me, i'll just replace these with that.

**Comment by tralezab** [comment]
@Cyberboss I handled your review and resolved the conflicts.

File: code/datums/mind.dm [hunk 10]
```diff
@@ -629,6 +629,31 @@
 	for(var/obj/effect/proc_holder/S in spell_list)
 		RemoveSpell(S)
 
+/datum/mind/proc/rot_mind() //most rotten spells also swap the victims spells for more rot spells, this proc handles that.
+	if(!spell_list.len)
+		return
+	var/spells_to_rot = 0 //how many rot spells to add.
+	for(var/obj/effect/proc_holder/spell/spell in spell_list)
+		spells_to_rot++
+		RemoveSpell(spell)
```
Comment (Shdorsh) [inline code/datums/mind.dm]: Add a can_rot var to spells, which is set to 0 (in order to automatically exclude racial spells). You will, sadly, have to set it to 1 for each spell that can rot individually. On this loop, you will create a list of spells that can rot and after that loop, RemoveSpell(pick(rotting_spells)) after a for loop depending on spells_to_rot. That way you can solve Ath's problem and also solve rotting spells removing rotting spells (which would be redundant).

Comment (tralezab) [reply code/datums/mind.dm]: Right, I did this but forgot to add it to the proc. Let me add that quick

**Comment by tralezab** [comment]
@ATH1909 Yeah its fantastic for downing demons. I might need to test interactions with different non wizard spells, but wizard rounds in general are chaos and end quickly so removing genetic powers, removing demon jaunts, it's all part of the fun as long as it doesn't... break... the game?

**Comment by tralezab** [comment]
more conflicts, waning interest because i've been looking at moods. maybe i'll come back to this later

**Comment by Kromgar** [comment]
@tralezab NOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO

**Comment by Shdorsh** [comment]
If I get enough time this weekend, mind if I pick up where you left?

**Comment by tralezab** [comment]
@Shdorsh absolutely do it. If you can get it merged I'd be happy to pr a corpulent demon after to finish off our nurgle party

**Comment by Shdorsh** [comment]
It's going to take some time though, my current byond project is a bit demanding and I need to finish some work from my internship.

**Pull request closed without merging**